### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by ikoote1**
Changes made :

1. Opened rails console for your project.
2. Created at least one user 
3. Created at least 4 posts written by one of the users
4. Created at least 6 posts comments for one of the posts
5. My models include some additional methods.
Users
- A method that returns the 3 most recent posts for a given user.

Posts

- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.

Comments
- A method that updates the comments counter for a post.

Likes
- A method that updates the likes counter for a post.

6. Opened a pull request
7

Thanks for reviewing my code

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (ishimwezachee) [inline app/models/user.rb]: - [x]  Good job on creating relation between tables but you still have something to improve on it . Based on the provided ERD diagram, your posts, comments and likes belong to users but olse they have foreign_key that connect them. Improve your codes by adding freign_key of authour_id to posts, comments and likes as well. 
eg:  `has_many : content, foreign_key : 'author_id'`

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :user
```
Comment (ishimwezachee) [inline app/models/post.rb:2]: - [OPTIONAL]  You can improve this line by saying `belongs_to :author, class_name:'User' ` to match the used names from the provided diagram.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (ishimwezachee) [inline app/models/comment.rb:2]: - [OPTIONAL]  You can improve this line by saying `belongs_to :author, class_name:'User' ` to match the used names from the provided diagram.

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (ishimwezachee) [inline app/models/like.rb:2]: - [OPTIONAL]  You can improve this line by saying `belongs_to :author, class_name:'User' ` to match the used names from the provided diagram.

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  has_many :likes
+
+  def recent_comments(limit = 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
+
+  def update_comments_counter
+    update(comments_counter: comments.count)
+  end
+
```
Comment (ishimwezachee) [inline app/models/post.rb]: - [x]  The methods used here are not correct, you should use proper methods names as we are doing methods for posts.  Check in the requirements : 
image

File: app/models/post.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  def update_likes_counter
+    update(likes_counter: likes.count)
+  end
```
Comment (ishimwezachee) [inline app/models/post.rb]: - [x]  This method belong to Like class, not post class just make sure each method provided is located in its corresponding class.

File: app/models/like.rb [hunk 7]
```diff
@@ -0,0 +1,4 @@
...
+  belongs_to :post
+end
```
Comment (ishimwezachee) [inline app/models/like.rb]: - [x]  As I said in post model, here should a method of updating likes counter  than putting it in Post class model and make sure it is working as it is supposed to.

File: app/models/comment.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
...
+  belongs_to :post
```
Comment (ishimwezachee) [inline app/models/comment.rb:3]: - [x]  Make sure you bring all the methods related to comment model class in here, and seemed like you've added them in your post class model.

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi @ikoote1 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image

## Highlights
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @ikoote1,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- No linter errors ✔️
- Correct workflow was used ✔️
- `well-organized and clean project ✔️
approv

# Status: Approved ✔️✔️✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- To write your commit messages in imperative mode and capitalize them. For example: Instead of writing only `correction` you can write `Correct linter errors`. This looks more professional.
error


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**